### DISCUSSION TITLE
Making RXJS optional

### DISCUSSION THREAD

**Discussion by KhaledGabr** [taiga-family/taiga-ui#10091] [Ideas]
Angular is heading into a direction where RXJS is going to be optional. Many apps doesn't need orchestrating complex asynchronous operations. Also Routes and Forms APIs will eventually use signals. 

https://www.youtube.com/watch?v=uOIdILz-Dkw

I know that Taiga heavily relies on RXJS internally, but can we have it so that library users don't have to import RXJS in their projects. 

In my code base, there is currently one file importing RXJS.

The TUI_DATE_FORMAT seems to expect an observable. 

import { of } from 'rxjs';
{ provide: TUI_DATE_FORMAT, useValue: of({ mode: 'YMD', separator: '/' }) },

Changing this will allow me to remove RXJS as a dependency from my project.

**Comment by waterplea** [comment]
Those things will likely become signals in next major release. I'm not sure it makes sense to introduce `signal | Observable` typing complexity until that.

**Reply by KhaledGabr** [reply]
Indeed, maintaining both simultaneously isn't worth the effort. 
Glad to hear that you have plans to moving things into Signals in future major releases!
